Tests/AuthenticationInterceptorTests.swift

```diff
@@ -447,10 +455,57 @@ class AuthenticationInterceptorTestCase: BaseTestCase {
         XCTAssertEqual(request.retryCount, 1)
     }
 
+    // Produces double lock reported in https://github.com/Alamofire/Alamofire/issues/3294#issuecomment-703241558
+    func testThatInterceptorNeedingImmediateRefreshWithSynchronouslyRefreshRefreshesRequestThatFailedWithOutdatedCredential() {
```

Comment:
This test case could likely use some simplification.

Reply:
Test seems fine to me overall. Here's my best stab at test name simplification.

```swift
func testThatInterceptorDoesNotDeadlockWhenAuthenticatorCallsRefreshCompletionSynchronouslyOnCallingQueue() {
```

Something like that?

If you have more specific concerns with the test, could you call them out? I'd be happy to dig deeper, but what you doing here makes sense to me and reproduces the reported issue. Also a great idea to add the issue that this test is designed for for reference 👌🏻